unsupported c macros: _IOR _IOW _IOWR etc.

Using Zig 0.7.0 with:

```zig
const c = @cImport(@cInclude("linux/videodev2.h"));
const std = @import("std");

pub fn main() !void {
  const fd = std.c.open("/dev/video0", std.c.O_RDWR);
  var cap :c.v4l2_capability = undefined;
  const ret = std.c.ioctl(fd, c.VIDIOC_QUERYCAP, &cap);
}
```

```
$ zig build-exe -lc main.zig
zig-cache/o/8bec9fe3242959012c1003557e26d644/cimport.zig:2019:29: error: unable to translate C expr: unexpected token .Keyword_struct
pub const VIDIOC_QUERYCAP = @compileError("unable to translate C expr: unexpected token .Keyword_struct"); // /nix/store/4j439q4svy053l6zx9z5fxww3qciq2gv-zig-0.7.0/lib/zig/libc/include/any-linux-any/linux/videodev2.h:2320:9
                            ^
./main.zig:7:32: note: referenced here
  const ret = std.c.ioctl(fd, c.VIDIOC_QUERYCAP, &cap);
```

There are a number of indirections in this macro, but ultimately it's something like 
```
( 1U << dirshift | 'V' << typeshift | nr << nrshift | sizeof(struct v4l2_capability) << sizeshift)
```

--- a/src/translate_c.zig
+++ b/src/translate_c.zig
@@ -5847,6 +5847,22 @@ fn parseCPrimaryExprInner(c: *Context, m: *MacroCtx, scope: *Scope) ParseError!*
         } else {
             return transCreateNodeIdentifierUnchecked(c, "c_int");
         },
+        .Keyword_enum, .Keyword_struct, .Keyword_union => {
+            // struct Foo will be declared as struct_Foo by transRecordDecl
+            const next_id = m.next().?;
+            if (next_id != .Identifier) {
+                try m.fail(c, "unable to translate C expr: expected Identifier instead got: {}", .{@tagName(next_id)});
+                return error.ParseError;
+            }
+
+            const ident_token = try appendTokenFmt(c, .Identifier, "{}_{}", .{slice, m.slice()});
+            const identifier = try c.arena.create(ast.Node.OneToken);
+            identifier.* = .{
+                .base = .{ .tag = .Identifier },
+                .token = ident_token,
+            };
+            return &identifier.base;
+        },
         .Identifier => {
             const mangled_name = scope.getAlias(slice);
             return transCreateNodeIdentifier(c, checkForBuiltinTypedef(mangled_name) orelse mangled_name);
@@ -5856,7 +5872,7 @@ fn parseCPrimaryExprInner(c: *Context, m: *MacroCtx, scope: *Scope) ParseError!*
 
             const next_id = m.next().?;
             if (next_id != .RParen) {
-                try m.fail(c, "unable to translate C expr: expected ')'' instead got: {}", .{@tagName(next_id)});
+                try m.fail(c, "unable to translate C expr: expected ')' instead got: {}", .{@tagName(next_id)});
                 return error.ParseError;
             }
             var saw_l_paren = false;
@@ -5886,7 +5902,7 @@ fn parseCPrimaryExprInner(c: *Context, m: *MacroCtx, scope: *Scope) ParseError!*
             const node_to_cast = try parseCExpr(c, m, scope);
 
             if (saw_l_paren and m.next().? != .RParen) {
-                try m.fail(c, "unable to translate C expr: expected ')''", .{});
+                try m.fail(c, "unable to translate C expr: expected ')'", .{});
                 return error.ParseError;
             }
 

--- a/test/stage1/behavior/translate_c_macros.h
+++ b/test/stage1/behavior/translate_c_macros.h
@@ -10,3 +10,9 @@ typedef struct Color {
 
 #define MY_SIZEOF(x) ((int)sizeof(x))
 #define MY_SIZEOF2(x) ((int)sizeof x)
+
+struct Foo {
+    int a;
+};
+
+#define SIZE_OF_FOO sizeof(struct Foo)

--- a/test/stage1/behavior/translate_c_macros.zig
+++ b/test/stage1/behavior/translate_c_macros.zig
@@ -16,3 +16,7 @@ test "sizeof in macros" {
     expectEqual(@as(c_int, @sizeOf(u32)), h.MY_SIZEOF(u32));
     expectEqual(@as(c_int, @sizeOf(u32)), h.MY_SIZEOF2(u32));
 }
+
+test "reference to a struct type" {
+    expectEqual(@sizeOf(h.struct_Foo), h.SIZE_OF_FOO);
+}

